### ISSUE TITLE
Basic types (i.e. String, Bool, etc) that are properties on an annotated type seem to inherit those annotations

### ISSUE THREAD

**Issue by mhays118** [krzysztofzablocki/Sourcery#972]
Hello,

I currently have a use-case that is requiring me to rename some types in my code (Objective-C wrappers around Swift code), and ran into a bit of an issue.

I want to add a suffix to all types that share the annotations. This was going well, until I hit a bit of a snag.

It appears that all basic properties on an annotated type inherit those annotations. As an example, if I have two models, a `User` and a `Color`, where a `User` has a favorite `Color`:

```swift
// sourcery: TestAnnotations
struct User {
    var id: String { return "test" }
    var favoriteColor: Color { return Color() }
}

// sourcery: TestAnnotations
struct Color {
    var rgb: String { return "000000" }
}
```

And the following stencil (naive/basic implementation to make it more readable):
```stencil
{% macro testAnnotate type %}{% if type|annotated:"TestAnnotations" %}Objc{% endif %}{% endmacro %}

{% for type in types.types where type|annotated:"TestAnnotations" %}
class {{ type.name }}Objc {
    {% for variable in type.allVariables %}
    var {{ variable.name }}: {{ variable.typeName.name }}{% call testAnnotate variable.type %} { value.{{ variable.name }} }
    {% endfor %}
}
{% endfor %}
```

This stencil will just create new classes with the same properties (read-only) and rename each annotated object to have an "Objc" appended to it's name.

Unfortunately, it appears to also do this to non-annotated types as well, as seen in the output:
```swift
class ColorObjc {
    var rgb: StringObjc { value.rgb }
}
class UserObjc {
    var id: StringObjc { value.id }
    var favoriteColor: ColorObjc { value.favoriteColor }
}
```

Am I perhaps missing something with the way annotations work?

---------

I tried adding a bit more, with a non-annotated type and that interestingly shows up as non-annotated, as expected. Maybe this is just something with more basic types? Or ones where they are not available in the source?

Source:
```swift 
// sourcery: TestAnnotations
struct User {
    var id: String { return "test" }
    var favoriteColor: Color { return Color( )}
    var other: UnannotatedType { return UnannotatedType() }
    var b: Bool = true
}

// sourcery: TestAnnotations
struct Color {
    var rgb: String { return "000000" }
}

struct UnannotatedType {
    var id: String { return "" }
}
```
Generates:
```swift
class ColorObjc {
    var rgb: StringObjc { value.rgb }
}
class UserObjc {
    var id: StringObjc { value.id }
    var favoriteColor: ColorObjc { value.favoriteColor }
    var other: UnannotatedType { value.other }
    var b: BoolObjc { value.b }
}

```

---------

Machine: MacBook Pro (15-inch, 2018)
OS: Mac OS 11.4 (20F71) (Big Sur)
Sourcery 1.4.2 installed via Mint

**Comment by krzysztofzablocki** [comment]
hey, 
Sourcery doesn't know about primitive types like `String`, `Int` since they aren't in your codebase, you can only use logic that depends on `typeName` for those, your type for primitives is nil unless you added some extensions.

This template seems a bit odd but you are probably trying to execute logic against undefined nil value, do a check if type is set before checking for annotations

ps. your stencil also has `types.types` which I assume is typo from `types.all` or `types.classes` ?

**Comment by mhays118** [comment]
Thanks @krzysztofzablocki!

Adding a nil check appears to have worked. Good to know that Sourcery doesn't know about primitive types! Thanks for your help and the quick response!

----

And for the `types.types` I actually saw it done like that in one of the included templates, so I followed suit. https://github.com/krzysztofzablocki/Sourcery/blob/d45b28715b2cb8a36285507cb14476e4c6072b26/Templates/Templates/AutoHashable.stencil#L12

It appears to work properly if I do either `types.types` or `types.all`. I'll go ahead and change my more complicated stencil to have `types.all` though. Thanks again!
